src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/InputAllowNonOverlappingLocalVars.java

```diff
@@ -0,0 +1,49 @@
+/*
+Input test file for RequireThisCheck.
+Created: 2017
+ */
+
+package com.puppycrawl.tools.checkstyle.checks.coding;
+
+class InputAllowNonOverlappingLocalVars {
+
+    private String s1 = "foo1";
+    String s2 = "foo2";
+
+    InputAllowNonOverlappingLocalVars() {
+        s1 = "bar1"; // Violation. Requires "this".
+        String s2;
+        s2 = "bar2"; // No violation. Non-overlapping overshadow by local var allowed.
```

Comment:
It is not clear how `overlapping` is different from `overshadowing`. This should be explained somewhere like in check's javadoc.